lib/Infer/Interpreter.cpp

```diff
@@ -426,6 +426,13 @@ namespace souper {
       else
         return {ARG0.getHiBits(1).trunc(1)};
     }
+    case Inst::Freeze: {
+      if (Args[0].K == EvalValue::ValueKind::Undef || Args[0].K == EvalValue::ValueKind::Poison) {
+        return {llvm::APInt(Args[0].BitWidth,
+                std::rand() % llvm::APInt::getAllOnesValue(Args[0].BitWidth).getLimitedValue())};
```

Comment:
is this the right thing to do? it seems like this will make testing non-deterministic, etc.

Reply:
Yes, this will make testing hard to do. But I don't think it's right idea either to make `freeze` always come up with same result?

Reply:
Well, this needs to be thought about carefully, since it could have an impact on soundness. Talk to @manasij7479 about it please.

Reply:
Not really sure what to do here, other than: https://xkcd.com/221/  
Maybe a 'freeze fuzzing' mode which runs the tests with freeze producing arbitrary results?

Reply:
what all are we using the interpreter for? the solution we choose has to be appropriate for that.

Reply:
We interpret both sides to see if they can produce a different result.
If all freezes produce the same result, this technique will not rule out some invalid transformations, which will then be sent off to the solver.
Similarly for pruning, we interpret the LHS and abstract-interpret the RHS.
It seems to me that producing the same value will (rarely, if at all) inhibit pruning, not introduce unsoundness.